### PR TITLE
Add extra parameters to toJSON method

### PR DESCRIPTION
**Pull request by vspinu**
Adding `preserve_proto_field_names = FALSE, always_print_primitive_fields = FALSE` parameters as per https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.util.json_util#JsonPrintOptions

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Seems sensible and low-touch.  As you know, we generally prefer a little reasoning and discussion before throwing unsolicited PRs over the wall.  Would you be able to show a quick JSON print "before" and "after"?

The package already supports use of `options()` for other toggles so I may put that on top.  The PR looks fine at a very first, very superficial glance and I appreciate that you added the motivation in the help page.

**Comment by vspinu** [comment]
> Would you be able to show a quick JSON print "before" and "after"?

```R
> p <- new(tutorial.Person, name = "Romain", id = 1,
+          phone = new(tutorial.Person.PhoneNumber, number = "12344"))
> 
> cat(p$toJSON())
{
 "name": "Romain",
 "id": 1,
 "phone": [
  {
   "number": "12344"
  }
 ]
}
> cat(p$toJSON(always_print_primitive_fields = T))
{
 "name": "Romain",
 "id": 1,
 "email": "",
 "phone": [
  {
   "number": "12344",
   "type": "HOME"
  }
 ]
}
> 
```

`preserve_proto_field_names` is just converting `snake_case` fields into `CamelCase`. IMO it should be TRUE by default but I makde it FALSE as that's the default in protobufc and python. 

Regarding options, I think it's an overkill for this case but if you really want I can add.

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,7 @@ src/*.dll
 GPATH
 GRTAGS
 GTAGS
-*~
\ No newline at end of file
+*~
+config.log
+config.status
+src/Makevars
```
Comment (eddelbuettel) [inline .gitignore]: These three transient files are removed by `./cleanup` which is why we survived 10+ years without them here.  

Most people I work with naturally keep PRs minimal.  Here I also happen to think it is better not to have them listed as it makes it clearer that the repo is _not_ clean'ed when one sees them.

It's minor but I like it that way so I restored it.

File: R/00classes.R [hunk 2]
```diff
@@ -189,7 +189,11 @@ setMethod("$", "Message", function(x, name) {
 		"set" = function(...) set( x, ... ),
 		"fetch" = function(...) fetch(x, ... ),
 		"toString" = function(...) toString( x, ... ),
-		"toJSON" = function(...) toJSON( x, ... ),
+		"toJSON" = function(preserve_proto_field_names = FALSE, always_print_primitive_fields = FALSE, ...)
+		  toJSON(x,
+				 preserve_proto_field_names = preserve_proto_field_names,
+				 always_print_primitive_fields = always_print_primitive_fields,
+				 ... ),
```
Comment (eddelbuettel) [inline R/00classes.R]: Indentation in this old file is all over place but never two chars.  I made it a little more resemblant of the existing form.

File: R/debug_string.R [hunk 3]
```diff
@@ -44,11 +44,14 @@ setMethod( "toString", "MethodDescriptor", ._toString_MethodDescriptor )
 setMethod( "toString", "FileDescriptor", ._toString_FileDescriptor )
 setMethod( "toString", "EnumValueDescriptor", ._toString_EnumValueDescriptor )
 
-setGeneric( "toJSON", function( x ){
+setGeneric( "toJSON", function( x, ... ) {
```
Comment (eddelbuettel) [inline R/debug_string.R:47]: Good catch, thanks.

**Review by eddelbuettel — APPROVED** [review]

**Comment by eddelbuettel** [comment]
> Regarding options, I think it's an overkill for this case but if you really want I can add.

I think it is nice to touch to control beghavior such as this via options but that is marginal and possibly only used by you so <shrug>.

File: R/00classes.R (hunk 2 above)
Comment (vspinu) [reply R/00classes.R]: It's not 2 spaces, that entire paragraph is indented with tabs, so I kept it this way for consistency. Different people and github will see different things. Everything else in the repo seem to be indented with spaces.

Comment (eddelbuettel) [reply R/00classes.R]: I use the same editor I always used (and which you use too, as I recall) and it came up wrong as it at GitHub making a 2:0 score which is why I fixed it. 

It is an old project and different tools where used by different people at different times. I do not believe in wholesale reformatting and in general tend to remove mode-lines now and rely mostly on `.editorconfig` -- which I just noticed had not been copied into this repo.  I may change that.

Comment (eddelbuettel) [reply R/00classes.R]: Done.  Appears to be a little better now, at least with an editor that groks it.

**Pull request merged**